File: backend/src/main/java/com/aifinancial/clarity/poc/service/AdminService.java
package com.aifinancial.clarity.poc.service;

import com.aifinancial.clarity.poc.dto.request.RoleUpdateRequest;
import com.aifinancial.clarity.poc.dto.response.MessageResponse;
import com.aifinancial.clarity.poc.dto.response.UserResponse;

import java.util.List;

public interface AdminService {
    /**
     * 獲取所有用戶列表
     * @return 用戶列表
     */
    List<UserResponse> getAllUsers();

    /**
     * 更新用戶角色
     * @param request 角色更新請求
     * @return 操作結果消息
     */
    MessageResponse updateUserRole(RoleUpdateRequest request);
} 

File: frontend/src/services/backend/adminService.ts
import { apiClient } from "./apiClient";
import type { MessageResponse, RoleUpdateRequest, UserResponse } from "./types";

/**
 * 管理员服务 - 负责与后端API交互进行管理员特定的操作
 * 这些API端点只能由具有Super Admin角色的用户访问
 */
export const adminService = {
	/**
	 * 获取所有用户列表
	 * @returns 用户数组
	 */
	getAllUsers: (): Promise<UserResponse[]> => {
		return apiClient.get<UserResponse[]>("/admin/users");
	},

	/**
	 * 更新用户角色
	 * @param userId 用户ID
	 * @param role 要设置的角色名称
	 * @returns 操作响应消息
	 */
	updateUserRole: (userId: number, role: string): Promise<MessageResponse> => {
		const request: RoleUpdateRequest = {
			userId,
			role,
		};
		return apiClient.post<MessageResponse>("/admin/users/role", request);
	},
};
